File: GMUtilsAndroid/GMUtils/src/main/java/com/blogspot/gm4s1/gmutils/storage/GeneralStorage.java
package com.blogspot.gm4s1.gmutils.storage;

import android.content.Context;
import android.content.SharedPreferences;
import android.text.TextUtils;

/**
 * Created by Ahmed El-Sayed (Glory Maker)
 * Computer Engineer / 2012
 * Android/iOS Developer with (Java/Kotlin, Swift)
 * Have experience with:
 *      - (C/C++, C#) languages
 *      - .NET environment
 *      - AVR Microcontrollers
 * a.elsayedabdo@gmail.com
 * +201022663988
 */
public class GeneralStorage {

    public static GeneralStorage getInstance() {
        return new GeneralStorage(null);
    }

    public static GeneralStorage getInstance(String name) {
        return new GeneralStorage(name);
    }

    /*private static String getPrefName() {
        return GeneralStorage.class.getName() + "GENERAL";
    }*/

    //----------------------------------------------------------------------------------------------

    private final SharedPreferences mPreference;

    private GeneralStorage(String name) {
        Context appContext = StorageManager.getAppContext();

        if (TextUtils.isEmpty(name)) {
            name = appContext.getPackageName() + "_" + getClass().getSimpleName().toUpperCase();
        } else {
            name = appContext.getPackageName() + "_" + name;
        }

        mPreference = appContext.getSharedPreferences(name, Context.MODE_PRIVATE);

    }

    public void save(String key, String value) {
        mPreference.edit().putString(key, value).apply();
    }

    public String retrieve(String key, String defaultValue) {
        return mPreference.getString(key, defaultValue);
    }

    public void remove(String key) {
        mPreference.edit().remove(key).apply();
    }

    public void removeAll() {
        mPreference.edit().clear().apply();
    }
}

File: GMUtilsFlutter/tutorial/lib/zgmutils/data_utils/storages/general_storage.dart
//shared_preferences: ^2.0.7
import 'dart:async';

import 'package:shared_preferences/shared_preferences.dart' as sharedPrefLib;

abstract class IStorage {

  Future<bool> save(
    String key,
    String data
  );

  //----------------------------------------------------------------------------

  Future<String?> retrieve(String key);

  //----------------------------------------------------------------------------

  Future<bool> remove(String key);

}

class GeneralStorage extends IStorage {
  final String prefName;

  GeneralStorage.o(this.prefName);

  sharedPrefLib.SharedPreferences? __prefs;

  Future<sharedPrefLib.SharedPreferences> get _prefs async {
    __prefs ??= await sharedPrefLib.SharedPreferences.getInstance();
    return __prefs!;
  }

  //----------------------------------------------------------------------------

  Future<bool> save(
    String key,
    String data,
  ) async {
    var pref = await _prefs;
    var b1 = await pref.setString('${prefName}_$key', data);
    return b1;
  }

  //----------------------------------------------------------------------------

  Future<String?> retrieve(String key) async {
    var pref = await _prefs;
    var data = pref.getString('${prefName}_$key');
    return data;
  }

  //----------------------------------------------------------------------------

  Future<bool> remove(String key) async {
    var pref = await _prefs;
    var b1 = await pref.remove('${prefName}_$key');
    return b1;
  }

}
